src/WarcraftLegacies.Source/Setup/Spells/SharedSpellSetup.cs

```diff
@@ -24,6 +24,11 @@ public static void Setup()
         },
         Effect = @"Abilities\Spells\Undead\ReplenishMana\SpiritTouchTarget.mdl"
       });
+
+      var tradingPost = new ProvidesIncome(Constants.UNIT_H014_TRADING_POST_SEA, 100);
```

Comment:
I think we're going to have to think through this implementation a bit more. `ProvidesIncome` is for bonus non-CP income, whereas this Trading Post is now operating just like a Control Point which makes it normal income. I think it probably needs to be integrated into the core Control Point system (even if they don't have CP towers).

Could you revert the Trading Post changes here? I'll spin off a new ticket to deal with those.

Reply:
ok, how would you want me to revert them, everything?

Reply:
Ah sorry I didn't notice this comment, yup what you've done is good.